src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+ * call getAbsolutePositionOfMainObject on the util object with the values from the two gear
+ * 
+ */
+public class CRTGearRatioUtil {
+  private final double mainGearRotationRatioTerm;
+  private final double mainGearRotationRatioTermSimplified;
+  private final double secondaryGearRotationRatioTermSimplfied;
+  private final double secondaryGearRotationRatioTerm;
+  private final int maxRotations;
+  private final double mainGearCircumference;
+
+  /**
+   * 
+   * @param mainGearRotationRatioTerm      for every mainGearRotationRatioTerm
+   *                                       rotations of the main gear,
+   *                                       the secondary gear rotates
+   *                                       secondaryGearRotationRatioTerm times
+   * @param secondaryGearRotationRatioTerm for every mainGearRotationRatioTerm
+   *                                       rotations of the main gear,
+   *                                       the secondary gear rotates
+   *                                       secondaryGearRotationRatioTerm times
+   * @param maxRotations                   the maximum number of rotations that
+   *                                       the main gear will make that involve
+   *                                       unique positions of the object
+   *                                       attached to it
+   * @param mainGearCircumference          circumference of the main gear
+   * 
+   * @return Returns optional of the util if the gear ratio works with the number
+   *         of max rotations, otherwise returns an empty optional
+   */
+  public static Optional<CRTGearRatioUtil> init(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm,
+      int maxRotations, double mainGearCircumference) {
+    CRTGearRatioUtil u = new CRTGearRatioUtil(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm, maxRotations,
+        mainGearCircumference);
+    if (u.validateGearRatioWithMaxRotations()) { //check if the gear ratio works with the required amount of rotations by the main gear
+      return Optional.of(u);
+    }
+    return Optional.empty();
+  }
+
+  private CRTGearRatioUtil(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm, int maxRotations,
+      double mainGearCircumference) {
+    this.mainGearRotationRatioTerm = (double) mainGearRotationRatioTerm;
+    this.secondaryGearRotationRatioTerm = (double) secondaryGearRotationRatioTerm;
+    this.mainGearRotationRatioTermSimplified = (double) (mainGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.secondaryGearRotationRatioTermSimplfied = (double) (secondaryGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.maxRotations = maxRotations;
+    this.mainGearCircumference = mainGearCircumference;
+  }
+
+  //"These two functions speak for themselves" - Hans Niemann commenting his code
```

Comment:
What does gcd stand for. Also can probably be private? Or leave it public and put some unit tests for it.

Reply:
I made it private. Is gcd being greatest common divisor not common knowledge? I'll add a comment, I guess.

Reply:
All acronyms should be specified. My work coding standard actually just straight up bans acronyms but I think this is fine with a comment